### PR TITLE
Introduce Recommended access order (RAO) via EA

### PR DESCRIPTION
**Pull request by y-sasn**
# Summary of changes

Introduce Recommended access order (RAO) support via EA by `ltfs.vendor.IBM.rao`.

# Description

This update introduce the support of the RAO (GRAO, RRAO) command, which was introduced from LTO-9.
Only linux scsi generic (sg) driver is supported in this fix.

## How to use

1. Set a file path to a correct GRAO Parameter Data via VEA on mount point.
2. A `xxx.out` file containing the RAO List will be made in the same directory.

example:
```
attr -s ltfs.vendor.IBM.rao -V ~/xxx <mount dir>
# The result will be created in `~/xxx.out`
```

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have confirmed my fix is effective or that my feature works

### CODE REVIEW

**Comment by piste-jp-ibm** [comment]
This is not good for merging to `v2.4-stable` because you already merged the changes to `master` on your branch.

I think it is impossible to merge your branch to `v2.4-stable` anymore. Could you create another PR to merge your changes to `master` ?

After that, I will try to merge it to `v2.4-stable` by cherry-pick.

**Review by piste-jp-ibm — CHANGES_REQUESTED** [review]
See my comment on conversation thread.

**Comment by piste-jp-ibm** [comment]
Can I close this one ?

**Comment by y-sasn** [comment]
Yes, I think we can. Closing PR.

File: src/tape_drivers/linux/lin_tape/lin_tape_ibmtape.c [hunk 1]
```diff
@@ -2228,23 +2229,13 @@ int lin_tape_ibmtape_logsense_page(void *device, const uint8_t page, const uint8
...
-{
-	int ret = 0;
-	struct lin_tape_ibmtape *priv = (struct lin_tape_ibmtape *) device;
-
-	ltfs_profiler_add_entry(priv->profiler, NULL, TAPEBEND_REQ_ENTER(REQ_TC_LOGSENSE));
-	ret = lin_tape_ibmtape_logsense_page(device, page, 0, buf, size);
 	ltfs_profiler_add_entry(priv->profiler, NULL, TAPEBEND_REQ_EXIT(REQ_TC_LOGSENSE));
-	return ret;
+
+	return logpage.len;
```
Comment (sevenever) [inline src/tape_drivers/linux/lin_tape/lin_tape_ibmtape.c]: Before this change, the ret value of lin_tape_ibmtape_logsense() is actually the ret code of ioctl SIOC_LOG_SENSE10_PAGE. With this change, the ret value was changed to length of data in buffer, which dooms to fail the check below in [lin_tape_ibmtape_remaining_capacity()](https://github.com/LinearTapeFileSystem/ltfs/pull/287/commits/1fbde528e7aa0ce2a253fbd8511dd91064c66cd6#diff-72823ff6a06c061d93257e41fe1a0436acb899d0b7872f8d75372cd119f4f134R3389), even if the IOCTL is actually succeed.
We are hitting "LTFS30457I Cannot get remaining capacity: get log page 0x17 failed ..." error on an IBM tape autoloader. I think this is probably the root cause?

**Pull request closed without merging**